scipy/sparse/linalg/_special_sparse_arrays.py

```diff
@@ -511,3 +513,419 @@ def _adjoint(self):
+        d2 = np.ones(self.n, dtype=self.dtype)
+        return spdiags([d2, d1, d0, d1, d2], [-2, -1, 0, 1, 2], self.n, self.n)
+
+    def tobanded(self):
+        """
+        Construct a Sakurai matrix in various formats and its eigenvalues.
+        """
+        d0 = np.r_[5, 6 * np.ones(self.n - 2, dtype=self.dtype), 5]
+        d1 = -4 * np.ones(self.n, dtype=self.dtype)
+        d2 = np.ones(self.n, dtype=self.dtype)
+        return np.array([d2, d1, d0])
+
+    def toarray(self):
+        d0 = np.r_[5, 6 * np.ones(self.n - 2, dtype=self.dtype), 5]
+        d1 = -4 * np.ones(self.n - 1, dtype=self.dtype)
+        d2 = np.ones(self.n - 2, dtype=self.dtype)
+        a = np.diag(d0)
+        a += np.diag(d1, 1) + np.diag(d1, -1)
+        a += np.diag(d2, 2) + np.diag(d2, -2)
+        return a
+    
+    def _matvec(self, x):
+        """
+        Construct matrix-free callable banded-matrix-vector multiplication by
+        the Sakurai matrix without constructing or storing the matrix itself
+        using the knowledge of its entries and the 5-diagonal format.
+        """
+        n = self.n
+        assert n == x.shape[0]
+        x = x.reshape(n, -1)
+        sx = np.zeros_like(x)
+        sx[0,:] = 5 * x[0,:] - 4 * x[1,:] + x[2,:]
+        sx[-1,:] = 5 * x[-1,:] - 4 * x[-2,:] + x[-3,:]
+        sx[1:-1,:] = (6 * x[1:-1,:] - 4 * (x[:-2,:] + x[2:,:])
+                      + np.pad(x[:-3,:], ((1,0),(0,0)))
+                      + np.pad(x[3:,:], ((0,1),(0,0))))
+        return sx
+
+    def _matmat(self, x):
+        """
+        Construct matrix-free callable matrix-matrix multiplication by
+        the Sakurai matrix without constructing or storing the matrix itself
+        by reusing the ``_matvec(x)`` that supports both 1D and 2D arrays ``x``.
+        """        
+        return self._matvec(x)
+
+    def _adjoint(self):
+        return self
+
+    def _transpose(self):
+        return self
+
+
+class MikotaM(LinearOperator):
+    """
+    Construct a mass matrix in various formats of Mikota pair.
+
+    The mass matrix `M` is square real diagonal
+    positive definite with entries that are reciprocal to integers.
+
+    Parameters
+    ----------
+    shape : tuple of int
+        The shape of the matrix.
+    dtype : dtype
+        Numerical type of the array. Default is ``np.float64``.
+
+    Methods
+    -------
+    toarray()
+        Construct a dense array from Mikota data
+    tosparse()
+        Construct a sparse array from Mikota data
+    tobanded()
+        The format for banded symmetric matrices,
+        i.e., (1, n) ndarray with the main diagonal.
+    """
+    def __init__(self, shape, dtype=np.float64):
+        self.shape = shape
+        self.dtype = dtype
+        super().__init__(dtype, shape)
+        # The matrix is constructed from its diagonal 1 / [1, ..., N+1];
+        # we precompute this to avoid duplicating the computation
+        self._diag = (1. / np.arange(1, shape[0] + 1)).astype(dtype)
```

Comment:
As mentioned, it's possible to make this a member function, then we don't "waste" the storage for the vector.
```suggestion
    def _diag(self):
        # The matrix is constructed from its diagonal 1 / [1, ..., N+1];
        # compute in a function to avoid duplicated code & storage footprint
        return (1. / np.arange(1, self.shape[0] + 1)).astype(self.dtype)
```